### PR TITLE
fix(SharingDetailsView): Prevent illegal unselection of read permissions

### LINKED ISSUE #46195: [Bug]: No error/warning when sharing a folder to another user without READ permission
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When a user shares a folder to another user in Nextcloud and removes READ permission when creating the share, the share is successfully created with READ permission but no warning or error is shown to the user.
User might think that READ permission is disabled but it is indeed enabled for that folder.

### Steps to reproduce

1. User A shares folder A with user B and removes READ permission when creating the share

### Expected behavior

The share is not created and there is an error saying something like "READ permission is needed for this share"
or
The share is created but there is a warning saying something like "READ permission has been added because it is needed for this share"

### Installation method

None

### Nextcloud Server version

28

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by nfebe**
## Summary 
With the exception of "file drop" on link shares all other shares need the read permissions at least.

Resolves : https://github.com/nextcloud/server/issues/46195

| Before (everywhere) | After (Folder link share) | Folder (Internal share, others|
| ------------------------------ | --------------------------------- | -------------------------------------------|
|Screenshot from 2024-07-05 15-52-24 | Screenshot from 2024-07-05 15-51-51-read-only-link | Screenshot from 2024-07-05 15-51-34 0 - read-only-internal |

### CODE REVIEW

File: apps/files_sharing/src/views/SharingDetailsTab.vue [hunk 1]
```diff
@@ -602,6 +602,9 @@ export default {
 			// allowed to revoke it too (but not to grant it again).
 			return (this.fileInfo.canDownload() || this.canDownload)
 		},
+		canRemoveReadPermission() {
+			return this.allowsFileDrop && this.share.type === this.SHARE_TYPES.SHARE_TYPE_LINK
+		},
```
Comment (artonge) [inline apps/files_sharing/src/views/SharingDetailsTab.vue:607]: I don't understand how the PR disables the checkbox? The condition is the same.

Comment (Fenn-CS) [reply apps/files_sharing/src/views/SharingDetailsTab.vue:607]: Initial condition 
```php
!allowsFileDrop && share.type === SHARE_TYPES.SHARE_TYPE_LINK
``` 
New condition 
```php
this.allowsFileDrop && this.share.type === this.SHARE_TYPES.SHARE_TYPE_LINK
``` 

Which is actually 
```php
!(this.allowsFileDrop && this.share.type === this.SHARE_TYPES.SHARE_TYPE_LINK)
``` 

via 

```php
!canRemoveReadPermission
```

**Review by artonge — APPROVED** [review]

**Review by Pytal — APPROVED** [review]

**Comment by skjnldsv** [comment]
Read permissions are also removable on email shares :fearful:

**Pull request merged**